tests/test_time_sync.py

```diff
@@ -93,3 +115,14 @@ def test_mocked(sensors: SensorSet, updater: TimeSyncUpdater, mock_adjtimex) ->
     assert sensors["ntp.state"].value == ClockState.OK
     for sensor in sensors.values():
         assert sensor.timestamp == 1234567890.654321
+
+
+def test_no_adjtimex(sensors: SensorSet, updater: TimeSyncUpdater, mock_no_adjtimex):
+    """Pretend that adjtimex is absent and test that it is handled correctly."""
+    updater.update()
+    assert isinstance(sensors["ntp.esterror"].value, float)
+    assert isinstance(sensors["ntp.maxerror"].value, float)
+    assert isinstance(sensors["ntp.state"].value, ClockState)
+    for sensor in sensors.values():
+        # Check that it actually got updated
+        assert sensor.status == Sensor.Status.INACTIVE
```

Comment:
A lot of code duplication here. I'd suggest either:
- Parametrise `test_smoke` with a boolean that says whether to mock, and conditionally use `mocker` inside the test; or
- Factor out a helper function that does all the work of the test, and have the test functions call it.

Reply:
I've gone for option 1 and lost 20 lines.